jbi/steps.py

```diff
@@ -3,21 +3,25 @@
 
 Each step takes an `ActionContext` and a list of arbitrary parameters.
 """
+from __future__ import annotations
```

Comment:
Can you explain why do we need this here? I thought importing everything necessary on top with `if TYPE_CHECKING` was enough

Maybe add a link to https://docs.python.org/3/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations